core/vm/jump_table_export.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package vm
+
+import "strings"
+
+// LookupInstructionSet returns the constructor for an instructionset for the
+// given fork.
+// If the fork is unknown, this method returns nil
+func LookupInstructionSet(fork string) func() JumpTable {
+	switch strings.ToLower(fork) {
+	case "frontier":
+		return newFrontierInstructionSet
+	case "homestead":
+		return newHomesteadInstructionSet
+	case "tangerinewhistle":
+		return newTangerineWhistleInstructionSet
+	case "spuriousdragon":
+		return newSpuriousDragonInstructionSet
+	case "byzantium":
+		return newByzantiumInstructionSet
```

Comment:
You missed petersburg :P

Reply:
We don't have any constructor for a petersburg - instructionset

Reply:
True, but that's an oddity. The pre-contantinople SSTORE gas calculation has the check for petersburg embedded in https://github.com/ethereum/go-ethereum/blob/master/core/vm/gas_table.go#L104, whereas Constantinople separated it out https://github.com/ethereum/go-ethereum/blob/master/core/vm/eips.go#L113.

Reply:
Interestingly, Istanbul should have net gas metering enabled and Petersburg is the one that disables it. So I guess the question is what the purpose of this public jump table is? How is it expected to refer to istanbul/petersburg?

Reply:
>  The pre-contantinople SSTORE gas calculation has the check for petersburg embedded in https://github.com/ethereum/go-ethereum/blob/master/core/vm/gas_table.go#L104, whereas Constantinople separated it out https://github.com/ethereum/go-ethereum/blob/master/core/vm/eips.go#L113.

Well, when making a change, there are two ways to do do it: 

1. modify the `op` to make it fork-aware, 
2. overwrite the `op` with a new, copied-and-modified `op`,  in the jumptable. 

I guess those petersburg/constantinople sstore-changes were done via `1`.

Reply:
> How is it expected to refer to istanbul/petersburg?

Well, normally, I'd expect to use only the latest, or "latest defined but not yet enabled", for fuzzing (more details below). Normally one would not fuzz ancent forks.

Reply:
But I agree, it makes more sense to enable all forks, and return the instructionset constructor regardless of whether it's a new one or an old one